src/https_client.c

```diff
@@ -316,14 +326,15 @@ void https_client_init(https_client_t *c, options_t *opt, struct ev_loop *loop)
 }
 
 void https_client_fetch(https_client_t *c, const char *url,
+                        const char* postdata, size_t postdata_len,
```

Comment:
FWIW, POST doesn't seem advised but I feel like caching (Read: MITM) should not be a thing over HTTPS so I'm totally OK with this. [GET support](https://tools.ietf.org/html/rfc8484#section-6) can be added by yourself or myself later.

[Seciton 5.1](https://tools.ietf.org/html/rfc8484#section-5.1)
```
   As a result, DoH servers need to carefully consider the HTTP caching
   metadata they send in response to GET requests (responses to POST
   requests are not cacheable unless specific response header fields are
   sent; this is not widely implemented and is not advised for DoH)
```

Reply:
Actually, now I re-read this, this paragraph is talking about server handling. Please disregard.